File: ios/Classes/ZegoUtils.m
//
//  ZegoUtils.m
//  Pods-Runner
//
//  Created by lizhanpeng@ZEGO on 2020/2/14.
//

#import "ZegoUtils.h"

@implementation ZegoUtils

+ (BOOL)boolValue:(NSNumber *)number {
    
    return [number isKindOfClass:[NSNull class]] ? NO : [number boolValue];
}

+ (int)intValue:(NSNumber *)number {
    
    return [number isKindOfClass:[NSNull class]] ? 0 : [number intValue];
}

+ (unsigned int)unsignedIntValue:(NSNumber *)number {
   
    return [number isKindOfClass:[NSNull class]] ? 0 : [number unsignedIntValue];
}

+ (long)longValue:(NSNumber *)number {
    return [number isKindOfClass:[NSNull class]] ? 0 : [number longValue];
}

+ (unsigned long)unsignedLongValue:(NSNumber *)number {
    
    return [number isKindOfClass:[NSNull class]] ? 0 : [number unsignedLongValue];
}

+ (float)folatValue:(NSNumber *)number {
    
    return [number isKindOfClass:[NSNull class]] ? 0.f : [number floatValue];
}

+ (double)doubleValue:(NSNumber *)number {
    
    return [number isKindOfClass:[NSNull class]] ? 0.0 : [number doubleValue];
}

@end

File: android/src/main/java/im/zego/zego_express_engine/internal/ZegoUtils.java
package im.zego.zego_express_engine.internal;

import java.util.ArrayList;
import java.util.Iterator;

public class ZegoUtils {
    /* Utils */

    public static boolean boolValue(Boolean number) {
        return number != null && number;
    }

    public static int intValue(Number number) {
        return number != null ? number.intValue() : 0;
    }

    public static long longValue(Number number) {
        return number != null ? number.longValue() : 0;
    }

    public static float floatValue(Number number) {
        return number != null ? number.floatValue() : .0f;
    }

    public static double doubleValue(Number number) {
        return number != null ? number.doubleValue() : .0f;
    }

    public static int[] intArrayValue(ArrayList<Integer> integerArrayList) {
        int[] ret = new int[integerArrayList.size()];
        Iterator<Integer> iterator = integerArrayList.iterator();
        for (int i = 0; i < ret.length; i++) {
            ret[i] = iterator.next();
        }
        return ret;
    }

    public static float[] floatArrayValueFromDoubleArray(ArrayList<Double> floatArrayList) {
        float[] ret = new float[floatArrayList.size()];
        Iterator<Double> iterator = floatArrayList.iterator();
        for (int i = 0; i < ret.length; i++) {
            ret[i] = iterator.next().floatValue();
        }
        return ret;
    }

    public static String getStackTrace(Throwable e) {
        StringBuilder message = new StringBuilder();
        if (e != null) {
            message.append(e.getClass()).append(": ").append(e.getMessage()).append(" | ");
            StackTraceElement[] elements = e.getStackTrace();
            for (StackTraceElement stackTraceElement : elements) {
                message.append(stackTraceElement.toString()).append(" | ");
            }
        }
        return message.toString();
    }
}
